datumaro/plugins/synthetic_data/image_generator.py

```diff
@@ -0,0 +1,208 @@
+        )
+
+        return parser
+
+    def __init__(self, output_dir: str, count: int, shape: Tuple[int, int], model_path: str = None):
+        self._count = count
+        self._cpu_count = min(os.cpu_count(), self._count)
+        assert len(shape) == 2
+        self._height, self._width = shape
+
+        self._output_dir = output_dir
+        if not osp.exists(output_dir):
+            os.makedirs(output_dir)
+
+        self._weights = self._create_weights(IFSFunction.NUM_PARAMS)
+        self._threshold = 0.2
+        self._iterations = 200000
+        self._num_of_points = 100000
+
+        self._path = model_path if model_path is not None else os.getcwd()
+        download_colorization_model(self._path)
+        self._initialize_params()
+
+    def generate_dataset(self) -> None:
+        log.warning(
+            "Generation of '%d' 3-channel images with height = '%d' and width = '%d'",
+            self._count,
+            self._height,
+            self._width,
+        )
+        use_multiprocessing = sys.platform != "darwin" or sys.version_info > (3, 7)
+        if use_multiprocessing:
+            with Pool(processes=self._cpu_count) as pool:
+                params = pool.map(
+                    self._generate_category, [Random(i) for i in range(self._categories)]
+                )
+        else:
+            params = []
+            for i in range(self._categories):
+                param = self._generate_category(Random(i))
+                params.append(param)
+
+        instances_weights = np.repeat(self._weights, self._instances, axis=0)
+        weight_per_img = np.tile(instances_weights, (self._categories, 1))
+        repeated_params = np.repeat(params, self._weights.shape[0] * self._instances, axis=0)
+        repeated_params = repeated_params[: self._count]
+        weight_per_img = weight_per_img[: self._count]
+        assert weight_per_img.shape[0] == len(repeated_params) == self._count
+
+        splits = min(self._cpu_count, self._count)
+        params_per_proc = np.array_split(repeated_params, splits)
+        weights_per_proc = np.array_split(weight_per_img, splits)
+
+        generation_params = []
+        offset = 0
+        for i, (param, w) in enumerate(zip(params_per_proc, weights_per_proc)):
+            indices = list(range(offset, offset + len(param)))
+            offset += len(param)
+            generation_params.append((i, Random(i), param, w, indices))
+
+        if use_multiprocessing:
+            with Pool(processes=self._cpu_count) as pool:
+                pool.starmap(self._generate_image_batch, generation_params)
+        else:
+            for i, param in enumerate(generation_params):
+                self._generate_image_batch(*param)
+
+    def _generate_image_batch(
+        self, job_id: int, rng: Random, params: np.ndarray, weights: np.ndarray, indices: List[int]
+    ) -> None:
+        print(f"{job_id}: started")
```

Comment:
I assume these `print`s were added to aid with debugging, but please don't forget to remove them later.